src/dbus/abrt_problems2_entry.c

```diff
@@ -966,10 +966,16 @@ uid_t abrt_p2_entry_get_owner(AbrtP2Entry *entry,
     {
         g_set_error(error, G_DBUS_ERROR, G_DBUS_ERROR_IO_ERROR,
                     "Failed open dump directory");
-        return -1;
+        return 0;
     }
 
-    const uid_t uid = dd_get_owner(dd);
+    int ret = 0;
+    const uid_t uid = dd_get_owner(dd, &ret);
+
+    if (ret != 0)
+        g_set_error(error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                    "Failed to get owner of dump directory");
```

Comment:
Do you think consulting `strerror()` and adding the output to the message would be useful?